src/components/colors-legend.components.tsx

```diff
@@ -32,25 +39,33 @@ const ColorsLegend: FC<MapControlProps> = ({ position = 'topright' }) => {
   const renderRow = (route: string) => {
     const routeInfo = allRoutes.find(rt => rt.id_upstream === route)!
     const routeColors = colors[route]
+    const listItemTextClasses = `${legendClasses.legendText} ${isLegendCollapsed ? legendClasses.collapsedText : ''}`
 
     return (
       <ListItem className={legendClasses.legendRow} key={route}>
-        <ListItemIcon>
+        <ListItemIcon className={legendClasses.legendIcon}>
           <div style={{ borderColor: routeColors.segment }} className={legendClasses.routeMarker}>
             <span style={{ color: routeColors.marker }}>{routeInfo.name_concise}</span>
           </div>
         </ListItemIcon>
-        <ListItemText className={legendClasses.legendText} primary={routeInfo.name_long} />
+        <ListItemText className={listItemTextClasses} primary={routeInfo.name_long} />
         <ListItemSecondaryAction>
           <RemoveRouteButton routeId={route} onClick={removeRoute} />
         </ListItemSecondaryAction>
       </ListItem>
     )
   }
 
+  const collapseButtonIconClasses = `${isLegendCollapsed ? legendClasses.rotatedToggleIcon : ''}`
   return (
     <MapControl position={position}>
       <Paper className={legendClasses.legendContainer} elevation={3}>
+        <div className={legendClasses.collapseWrapper}>
+          <Button className={legendClasses.collapseButton} onClick={collapseLegend}>
+            <DoubleArrowRoundedIcon className={collapseButtonIconClasses} />
+          </Button>
+        </div>
```

Comment:
Maybe have a separate component like `RemoveRouteButton` ?

Reply:
@strdr4605 Did you mean `ToggleRoutesButton` instead of `RemoveRouteButton`?

Reply:
Yes, I just wrote “like remove...”